File: backend/app/schemas/task.py
from typing import Literal

from pydantic import BaseModel

TaskStatus = Literal["pending", "processing", "paused", "done", "failed"]
TaskItemStatus = Literal["pending", "uploading", "submitted", "processing", "paused", "done", "failed"]
TaskMode = Literal["single", "batch"]


class TaskItem(BaseModel):
    itemId: str
    fileKey: str
    fileName: str
    status: TaskItemStatus
    message: str = "等待处理"
    downloadUrl: str | None = None


class TaskDetail(BaseModel):
    taskId: str
    status: TaskStatus
    mode: TaskMode
    total: int
    completed: int
    failed: int
    logs: list[str]
    items: list[TaskItem]


class CreateTaskResponse(BaseModel):
    taskId: str
    total: int
    status: TaskStatus


class TaskListResponse(BaseModel):
    tasks: list[TaskDetail]


class ClearTasksResponse(BaseModel):
    deletedTasks: int
    deletedFiles: int


class DeleteTaskItemResponse(BaseModel):
    deletedTask: bool
    deletedItem: bool
    deletedFiles: int
    task: TaskDetail | None = None

File: frontend/src/types/task.ts
export type TaskStatus = 'pending' | 'processing' | 'paused' | 'done' | 'failed';
export type TaskItemStatus = 'pending' | 'uploading' | 'submitted' | 'processing' | 'paused' | 'done' | 'failed';
export type TaskMode = 'single' | 'batch';

export interface TaskItem {
  itemId: string;
  fileKey: string;
  fileName: string;
  status: TaskItemStatus;
  message: string;
  downloadUrl: string | null;
  originalUrl: string | null;
}

export interface TaskItemRecord {
  taskId: string;
  item: TaskItem;
}

export interface TaskDetail {
  taskId: string;
  status: TaskStatus;
  mode: TaskMode;
  total: number;
  completed: number;
  failed: number;
  logs: string[];
  items: TaskItem[];
}

export interface CreateTaskResponse {
  taskId: string;
  total: number;
  status: TaskStatus;
}

export interface TaskListResponse {
  tasks: TaskDetail[];
}

export interface ClearTasksResponse {
  deletedTasks: number;
  deletedFiles: number;
}

export interface DeleteTaskItemResponse {
  deletedTask: boolean;
  deletedItem: boolean;
  deletedFiles: number;
  task: TaskDetail | null;
}
